pinot-spi/src/main/java/org/apache/pinot/spi/plugin/PluginManager.java

```diff
@@ -111,13 +122,30 @@ public class PluginManager {
         }
       };
 
-  private Map<Plugin, PluginClassLoader> _registry = new HashMap<>();
+  private final ClassWorld _classWorld;
+  private final Map<Plugin, PluginClassLoader> _registry;
+
   private String _pluginsDirectories;
   private String _pluginsInclude;
   private boolean _initialized = false;
 
-  private PluginManager() {
-    _registry.put(new Plugin(DEFAULT_PLUGIN_NAME), createClassLoader(Collections.emptyList()));
+  PluginManager() {
+    if (useLegacyPluginClassloader()) {
```

Comment:
Can we add a log (in info level) saying whether we are using legacy mode or not? That is the kind of log that help a lot when having issues in production 😆